### PR TITLE
feat(component): Add chip component

### PR DESCRIPTION
**Pull request by animesh1987**
## What
- Add chip component.

## Testing
- Jest tests

@jorgemoya

### CODE REVIEW

File: packages/big-design/src/components/Chip/Chip.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const renderDeleteButton = () =>
+    onDelete && (
+      <StyledCloseButton
+        variant="subtle"
+        onClick={handleOnDelete}
+        iconOnly={<CloseIcon size="medium" title="Delete" />}
+      ></StyledCloseButton>
+    );
+
+  return (
```
Comment (chanceaclark) [inline packages/big-design/src/components/Chip/Chip.tsx]: Do we even want to render the chip if there is no `label`?
```suggestion
  return label ? (
```

**Comment by deini** [comment]
Hey @animesh1987  👋 pushed a commit with a couple of changes mainly:

- Changed the API to `<Chip>Text</Chip>`
- There is only one prop now, `onDelete` which we also use to display/hide the close button.
- The  close svg is now wrapped in a `Button` component for accessibility

File: packages/big-design/src/components/Chip/Chip.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      margin="xxSmall"
+      borderRadius="normal"
+    >
+      <Text margin="none" marginRight="xxSmall">
+        {label}
+      </Text>
+
+      {renderDeleteButton()}
+    </StyledChip>
+  );
```
Comment (chanceaclark) [inline packages/big-design/src/components/Chip/Chip.tsx]: Addition from above.
```suggestion
  ) : null;
```

File: packages/big-design/src/components/Chip/Chip.tsx (hunk 1 above)
Comment (chanceaclark) [reply packages/big-design/src/components/Chip/Chip.tsx]: If we add the above, you should add another test with no children.

File: packages/big-design/src/components/Chip/styled.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
+import styled from 'styled-components';
+
+import { Box } from '../Box';
+import { StyleableButton } from '../Button/Button';
+
+export const StyledChip = styled(Box)`
+  align-items: center;
+  display: inline-flex;
+  user-select: none;
```
Comment (chanceaclark) [inline packages/big-design/src/components/Chip/styled.tsx]: 🍹Is there a specific reason why we wouldn't want people to select the text?

Comment (deini) [reply packages/big-design/src/components/Chip/styled.tsx]: Not really, it just looks terrible and doesn't seem that useful 🤔

File: packages/big-design/src/components/Chip/Chip.tsx (hunk 1 above)
Comment (deini) [reply packages/big-design/src/components/Chip/Chip.tsx]: I thought about that but I _think_ it's a bit easier to spot if we render a chip which looks wrong than to not render at all. Thoughts?

Comment (jorgemoya) [reply packages/big-design/src/components/Chip/Chip.tsx]: Just from my own experience experimenting with `chips`, I think it's better if we display an empty chip than no chip at all. I was able to debug I wasn't sending the right text immediately.

**Comment by animesh1987** [comment]
Thanks for pushing it through @deini 👍

**Pull request merged**